### PR TITLE
Lusine: Factory Method Enemy Ship

### PR DESCRIPTION
**Pull request by hsp-lusine**

### CODE REVIEW

File: src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/EnemyShipFactory.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    /**
+     * This could be used as a static method if we are willing to give up
+     * subclassing it
+     * 
+     * @param newShipType
+     * @return
+     */
+    public static EnemyShip makeEnemyShip(String newShipType) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/EnemyShipFactory.java]: It is better to use primitive types rather than reference types...

File: src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/EnemyShipFactory.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        if (newShipType.equals("U")) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/EnemyShipFactory.java]: Remove magic strings...

File: src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/EnemyShip.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ *Abstract class, which responsible all generic methods and fields
+ * 
+ * @author lusine
+ *
+ */
+public abstract class EnemyShip {
+
+    protected String name;
+    protected double speed;
+    protected double directionX;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/EnemyShip.java]: do you use this 2 instance variables anywhere?

File: src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/BigUFOEnemyShip.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ * @author lusine
+ *
+ */
+public class BigUFOEnemyShip extends UFOEnemyShip {
+
+    /**
+     * Constructor for initializing instance variable
+     */
+    public BigUFOEnemyShip() {
+        name = "Big UFO Enemy Ship";
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/BigUFOEnemyShip.java]: remove magic numbers/strings...
You need to have appropriate constants for storing Default Values...

File: src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/RocketEnemyShip.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ * @author lusine
+ *
+ */
+public class RocketEnemyShip extends EnemyShip {
+
+    /**
+     * Constructor for initializing instance variable
+     */
+    public RocketEnemyShip() {
+        name = "Rocket Enemy Ship";
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/RocketEnemyShip.java]: same

File: src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/UFOEnemyShip.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+ * @author lusine
+ *
+ */
+public class UFOEnemyShip extends EnemyShip {
+
+    /**
+     * Constructor for initializing instance variable
+     */
+    public UFOEnemyShip() {
+        name = "UFO Enemy Ship";
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/UFOEnemyShip.java]: same

File: src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/shape/Circle.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
+package main.java.am.hsp.apps.oop.desingPattern.factoryMethod.shape;
+
+/**
+ * Contain all the methods for circle
+ * 
+ * @author lusine
+ *
+ */
+public class Circle extends Shape {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/shape/Circle.java]: These classes are not related to this Pull Request - as you have another PR for the task related shapes...

File: src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/EnemyShip.java (hunk 3 above)
Comment (hsp-lusine) [reply src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/EnemyShip.java]: i don't use instance variables(speed and coordinate), but i think which i must have a speed and coordinate for all objects.... if I don't understand  right,  please say....

File: src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/EnemyShipFactory.java (hunk 1 above)
Comment (hsp-lusine) [reply src/main/java/am/hsp/apps/oop/desingPattern/factoryMethod/enemyShip/EnemyShipFactory.java]: Pleaseeee , say a bit detailis
